Source/WebCore/dom/InternalObserverInspect.cpp

```diff
@@ -0,0 +1,252 @@
+        {
+            RefPtr context = scriptExecutionContext();
+
+            if (!context) {
+                subscriber.complete();
+                return { };
+            }
+
+            if (m_inspector.subscribe) {
+                auto* globalObject = protectedScriptExecutionContext()->globalObject();
+                ASSERT(globalObject);
+
+                Ref vm = globalObject->vm();
+
+                {
+                    JSC::JSLockHolder lock(vm);
+                    auto scope = DECLARE_CATCH_SCOPE(vm);
+
+                    m_inspector.subscribe->handleEventRethrowingException();
+
+                    JSC::Exception* exception = scope.exception();
+                    if (UNLIKELY(exception)) {
+                        scope.clearException();
+                        subscriber.error(exception->value());
+                        return { };
+                    }
+                }
+            }
+
+            SubscribeOptions options;
+            options.signal = &subscriber.signal();
+            m_sourceObservable->subscribeInternal(*context, InternalObserverInspect::create(*context, subscriber, m_inspector), options);
+
+            return { };
+        }
+
+        CallbackResult<void> handleEventRethrowingException(Subscriber& subscriber) final
+        {
+            return handleEvent(subscriber);
+        }
+
+    private:
+        bool hasCallback() const final { return true; }
+
+        SubscriberCallbackInspect(ScriptExecutionContext& context, Ref<Observable> source, const ObservableInspector& inspector)
+            : SubscriberCallback(&context)
+            , m_sourceObservable(source)
+            , m_inspector(inspector)
+        { }
+
+        Ref<Observable> m_sourceObservable;
+        ObservableInspector m_inspector;
+    };
+
+private:
+    void next(JSC::JSValue value) final
+    {
+        if (m_inspector.next) {
+            Ref vm = protectedGlobalObjectVM();
+            JSC::JSLockHolder lock(vm);
+            auto scope = DECLARE_CATCH_SCOPE(vm);
+
+            m_inspector.next->handleEventRethrowingException(value);
+
+            JSC::Exception* exception = scope.exception();
+            if (UNLIKELY(exception)) {
+                scope.clearException();
+                m_subscriber->error(exception->value());
+                return;
+            }
+        }
+
+        m_subscriber->next(value);
+    }
+
+    void error(JSC::JSValue value) final
+    {
+        removeAbortHandler();
+
+        if (m_inspector.error) {
+            Ref vm = protectedGlobalObjectVM();
+            JSC::JSLockHolder lock(vm);
+            auto scope = DECLARE_CATCH_SCOPE(vm);
+
+            m_inspector.error->handleEventRethrowingException(value);
+
+            JSC::Exception* exception = scope.exception();
+            if (UNLIKELY(exception)) {
+                scope.clearException();
+                m_subscriber->error(exception->value());
```

Comment:
Need a smart pointer to m_subscriber *on the stack* before calling the member function on it. The safer cpp bot should complain.